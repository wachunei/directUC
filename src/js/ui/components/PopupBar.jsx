import styled from "styled-components";

const PopupBar = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 0.5rem 0;

  &:first-of-type {
    margin-top: 0;
  }
  &:last-of-type {
    margin-bottom: 0;
  }
`;

export default PopupBar;
