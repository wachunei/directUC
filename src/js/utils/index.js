export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
