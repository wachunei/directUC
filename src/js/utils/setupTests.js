// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
