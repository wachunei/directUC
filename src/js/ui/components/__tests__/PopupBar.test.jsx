import React from "react";
import PopupBar from "../PopupBar";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("PopupBar Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<PopupBar />);
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with lightTheme", () => {
    const tree = renderTree(<PopupBar />, "lightTheme");
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with darkTheme", () => {
    const tree = renderTree(<PopupBar />, "darkTheme");
    expect(tree).toMatchSnapshot();
  });
});
