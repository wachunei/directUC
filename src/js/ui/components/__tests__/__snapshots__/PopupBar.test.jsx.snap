// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupBar Component should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0.5rem 0;
}

.c0:first-of-type {
  margin-top: 0;
}

.c0:last-of-type {
  margin-bottom: 0;
}

<div
  class="c0"
/>
`;

exports[`PopupBar Component should match snapshot with darkTheme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0.5rem 0;
}

.c0:first-of-type {
  margin-top: 0;
}

.c0:last-of-type {
  margin-bottom: 0;
}

<div
  class="c0"
/>
`;

exports[`PopupBar Component should match snapshot with lightTheme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0.5rem 0;
}

.c0:first-of-type {
  margin-top: 0;
}

.c0:last-of-type {
  margin-bottom: 0;
}

<div
  class="c0"
/>
`;
